crates/nodeapi/src/lib.rs

```diff
@@ -191,10 +219,13 @@ pub async fn service_health(
     }
 }
 
-/// Function to create the Actix HTTP server
+/// Function to create the Ntex HTTP server
 /// This function sets up the server and routes.
 /// External users can call this function to create and run the server.
-pub fn create_server(api: NodeApi, ip_port_addr: String) -> std::io::Result<ntex::server::Server> {
+pub fn create_server(
+    api: Arc<Mutex<NodeApi>>,
```

Comment:
Let's move `NodeApi` to include the `Arc` and `Mutex`. We can rename the inner type to `NodeApiInner` or smth like that.